test/jellyfish/component/hls/request_handler_test.exs

```diff
@@ -0,0 +1,148 @@
+defmodule Jellyfish.Component.HLS.RequestHandlerTest do
+  @moduledoc false
+
+  use ExUnit.Case, async: true
+
+  alias Jellyfish.Component.HLS
+  alias Jellyfish.Component.HLS.{EtsHelper, RequestHandler}
+
+  @wrong_room_id "321"
+
+  @manifest "index.m3u8"
+  @wrong_manifest "wrong.m3u8"
+  @manifest_content "manifest"
+
+  @partial {1, 1}
+  @next_partial {1, 2}
+  @partial_name "partial"
+  @partial_content <<1, 2, 3>>
+
+  @offset 0
+  @wrong_offset 1
+
+  setup do
+    room_id = UUID.uuid4()
+
+    # RequestHandler is not removed at all in tests
+    # It removes itself when parent process is killed
+    RequestHandler.start(room_id)
+
+    %{room_id: room_id}
+  end
+
+  test "room managment", %{room_id: room_id} do
+    {:error, {:already_started, _pid}} = RequestHandler.start(room_id)
+    {:ok, _table} = EtsHelper.add_room(room_id)
+
+    RequestHandler.stop(room_id)
+
+    # wait for ets to be removed
+    Process.sleep(200)
+
+    {:error, :room_not_found} = EtsHelper.get_manifest(room_id)
+
+    {:ok, _pid} = RequestHandler.start(room_id)
+  end
+
+  test "file request", %{room_id: room_id} do
+    add_mock_manifest(room_id)
+
+    {:ok, @manifest_content} = RequestHandler.handle_file_request(room_id, @manifest)
+
+    {:error, :enoent} = RequestHandler.handle_file_request(room_id, @wrong_manifest)
+    {:error, :enoent} = RequestHandler.handle_file_request(@wrong_room_id, @manifest)
+
+    remove_mock_manifest(room_id)
+  end
+
+  test "manifest request", %{room_id: room_id} do
+    {:error, :room_not_found} = RequestHandler.handle_manifest_request(room_id, @partial)
+
+    {:ok, table} = EtsHelper.add_room(room_id)
+
+    {:error, :file_not_found} = RequestHandler.handle_manifest_request(room_id, @partial)
+
+    EtsHelper.update_recent_partial(table, @partial)
+    EtsHelper.update_manifest(table, @manifest_content)
+    RequestHandler.update_recent_partial(room_id, @partial)
+
+    {:ok, @manifest_content} = RequestHandler.handle_manifest_request(room_id, @partial)
+
+    pid = self()
+
+    spawn(fn ->
+      {:ok, @manifest_content} = RequestHandler.handle_manifest_request(room_id, @next_partial)
+      send(pid, :manifest)
+    end)
+
+    refute_receive(:manifest, 500)
```

Comment:
I can't change it like in the previous example because handle_manifest_request is blocking

Reply:
I think in this case you should use Task instead of spawn, and instead of this strange sending message and refute receive, use [Task.yield](https://hexdocs.pm/elixir/1.13/Task.html#yield/2).